app/controllers/works_controller.rb

```diff
@@ -2,6 +2,7 @@ class WorksController < ApplicationController
   # We should always be able to tell what category
   # of work we're dealing with
   before_action :category_from_work, except: [:root, :index, :new, :create]
+  before_action :require_login, only: [:index, :show]
```

Comment:
Why not do this for the `create`, `edit`, `update`, `destroy` and `new` actions as well?